File: android/app/src/main/java/com/neostudios/zest/domain/usecase/NutritionCalculator.kt
package com.neostudios.zest.domain.usecase

import com.neostudios.zest.domain.model.Ingredient
import com.neostudios.zest.domain.model.NutritionInfo

object NutritionCalculator {
    private val nutritionDatabase = mapOf(
        "chicken breast" to NutritionInfo(165, 31.0, 0.0, 3.6),
        "rice" to NutritionInfo(130, 2.7, 28.0, 0.3),
        "broccoli" to NutritionInfo(55, 3.7, 11.0, 0.6),
        "olive oil" to NutritionInfo(884, 0.0, 0.0, 100.0),
        "egg" to NutritionInfo(155, 13.0, 1.1, 11.0),
        "milk" to NutritionInfo(61, 3.2, 4.8, 3.3),
        "bread" to NutritionInfo(265, 9.0, 49.0, 3.2),
        "pasta" to NutritionInfo(131, 5.0, 25.0, 1.1),
        "tomato" to NutritionInfo(18, 0.9, 3.9, 0.2),
        "cheese" to NutritionInfo(402, 25.0, 1.3, 33.0)
    )
    
    fun calculate(ingredients: List<Ingredient>, servings: Int): NutritionInfo {
        var totalCalories = 0.0
        var totalProtein = 0.0
        var totalCarbs = 0.0
        var totalFat = 0.0
        
        ingredients.forEach { ing ->
            val nutrition = nutritionDatabase[ing.name.lowercase()]
            if (nutrition != null) {
                val factor = ing.amount / 100.0
                totalCalories += nutrition.calories * factor
                totalProtein += nutrition.protein * factor
                totalCarbs += nutrition.carbs * factor
                totalFat += nutrition.fat * factor
            }
        }
        
        return NutritionInfo(
            calories = (totalCalories / servings).toInt(),
            protein = (totalProtein / servings * 10).toInt() / 10.0,
            carbs = (totalCarbs / servings * 10).toInt() / 10.0,
            fat = (totalFat / servings * 10).toInt() / 10.0
        )
    }
    
    fun estimateCost(ingredients: List<Ingredient>): Double {
        return (ingredients.size * 2.5 * 100).toInt() / 100.0
    }
}

File: ios/Zest/Services/NutritionCalculator.swift
import Foundation

struct NutritionInfo: Codable {
    let calories: Int
    let protein: Double
    let carbs: Double
    let fat: Double
}

class NutritionCalculator {
    static let shared = NutritionCalculator()
    
    private let nutritionDatabase: [String: NutritionInfo] = [
        "chicken breast": NutritionInfo(calories: 165, protein: 31, carbs: 0, fat: 3.6),
        "rice": NutritionInfo(calories: 130, protein: 2.7, carbs: 28, fat: 0.3),
        "broccoli": NutritionInfo(calories: 55, protein: 3.7, carbs: 11, fat: 0.6),
        "olive oil": NutritionInfo(calories: 884, protein: 0, carbs: 0, fat: 100),
        "egg": NutritionInfo(calories: 155, protein: 13, carbs: 1.1, fat: 11),
        "milk": NutritionInfo(calories: 61, protein: 3.2, carbs: 4.8, fat: 3.3),
        "bread": NutritionInfo(calories: 265, protein: 9, carbs: 49, fat: 3.2),
        "pasta": NutritionInfo(calories: 131, protein: 5, carbs: 25, fat: 1.1),
        "tomato": NutritionInfo(calories: 18, protein: 0.9, carbs: 3.9, fat: 0.2),
        "cheese": NutritionInfo(calories: 402, protein: 25, carbs: 1.3, fat: 33)
    ]
    
    func calculate(ingredients: [Ingredient], servings: Int) -> NutritionInfo {
        var totalCalories = 0.0
        var totalProtein = 0.0
        var totalCarbs = 0.0
        var totalFat = 0.0
        
        for ingredient in ingredients {
            if let nutrition = nutritionDatabase[ingredient.name.lowercased()] {
                let factor = ingredient.amount / 100.0
                totalCalories += Double(nutrition.calories) * factor
                totalProtein += nutrition.protein * factor
                totalCarbs += nutrition.carbs * factor
                totalFat += nutrition.fat * factor
            }
        }
        
        return NutritionInfo(
            calories: Int(totalCalories / Double(servings)),
            protein: round(totalProtein / Double(servings) * 10) / 10,
            carbs: round(totalCarbs / Double(servings) * 10) / 10,
            fat: round(totalFat / Double(servings) * 10) / 10
        )
    }
    
    func estimateCost(ingredients: [Ingredient]) -> Double {
        return round(Double(ingredients.count) * 2.5 * 100) / 100
    }
}
